src/util/ui/widgets.rs

```diff
@@ -28,58 +47,76 @@ impl<'w, 's> MyWidgets<'w> for Commands<'w, 's> {
                 ..default()
             },
             ..default()
-        })
+        });
+        entity_commands.with_children(spawn_children);
+        entity_commands
     }
+}
+
+/// Widgets spawning trait.
+pub(crate) trait Widgets<'a> {
+    /// Spawns a simple button node with text.
+    fn my_button<I: Into<String>>(&mut self, text: I) -> EntityCommands;
 
+    //// Spawns a simple label.
+    fn my_label<I: Into<String>>(&mut self, text: I) -> EntityCommands;
+}
+
+impl<'a, T: Spawner<'a>> Widgets<'a> for T {
```

Comment:
```suggestion
impl<'a, T> Widgets<'a> for T 
	where T: Spawner<'a>
{
```
Personal preference, feel free to disregard :)

Reply:
removed for widgets, my_root -> ui_root so it's a bit clearer what it is

Reply:
Hmm? I believe my suggestion still applies. It's just about the amount of noise that is happening on the line before we get to the name `Widgets`.